pkg/scheduler/objects/application.go

```diff
@@ -200,6 +204,17 @@ func (sa *Application) timeOutStarting() {
 	}
 }
 
+func (sa *Application) timeOutWaiting() {
+	if sa.IsWaiting() {
+		log.Logger().Warn("Application in waiting state timed out: marking it as completed",
+			zap.String("applicationID", sa.ApplicationID),
+			zap.String("state", sa.stateMachine.Current()))
```

Comment:
I don't think this is a WARN. We can change this to DEBUG safely

Reply:
Done